hal/armv8m/mcx/n94x/_init.S

```diff
@@ -64,6 +64,13 @@ _vtor_temp:
 .type _start, %function
 _start:
 	cpsid if
+#ifdef MCX_CPU0_RST_ADDR
+	ldr     r0, =MCX_CHECK_ADDR
```

Comment:
For future reference: we may want to switch to Mailbox peripheral for signalling the primary core (CPU0) that the secondary core (CPU1) has started. This way specifying fixed addresses `MCX_CPU0_RST_ADDR` and `MCX_CHECK_ADDR` will no longer be necessary.